File: src/studly.py
"""
Studly Caps Filter
==================

Randomly capitalizes letters to create "StUdLy CaPs" effect.

Author: Nick Phillips (original)
License: GPL-2+
"""

import random


def transform(text: str, seed: int = 42, **kwargs) -> str:
    """
    Transform text with random capitalization (studly caps).

    Args:
        text: Input text to transform
        seed: Random seed for reproducibility (default: 42)
        **kwargs: Additional parameters (ignored)

    Returns:
        Text with StUdLy CaPiTaLiZaTiOn
    """
    rng = random.Random(seed)
    result = []

    for char in text:
        if char.isalpha():
            # Randomly capitalize or lowercase
            if rng.random() < 0.5:
                result.append(char.upper())
            else:
                result.append(char.lower())
        else:
            result.append(char)

    return ''.join(result)

File: src/javascript/studly.js
/**
 * Studly Caps Filter
 * ===================
 *
 * Randomly capitalizes letters to create "StUdLy CaPs" effect.
 *
 * Author: Nick Phillips (original)
 * License: GPL-2+
 */

class StudlyTransformer {
    constructor(seed = 42) {
        this.random = new SeededRandom(seed);
    }

    transform(text) {
        const result = [];

        for (const char of text) {
            if (/[a-zA-Z]/.test(char)) {
                if (this.random.random() < 0.5) {
                    result.push(char.toUpperCase());
                } else {
                    result.push(char.toLowerCase());
                }
            } else {
                result.push(char);
            }
        }

        return result.join('');
    }
}
